### PR TITLE
Feature/main app

### PR DESCRIPTION
**Pull request by mathisfchx**
app.go (+go.mod)
Dockerfile
Readme

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# syntax=docker/dockerfile:1
+FROM golang:1.18.1
```
Comment (jeremyjams) [inline Dockerfile:2]: If it fits your use-case, I suggest to take the alpine version which is lighter
```suggestion
FROM golang:1.18.1-alpine
```
See weight difference by browsing https://hub.docker.com/_/golang?tab=tags&page=2

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+COPY go.mod ./
+
+RUN go mod download
+
+COPY src/*.go ./
+
+RUN go build -o /docker-gs-ping
+
+EXPOSE 8080
```
Comment (jeremyjams) [inline Dockerfile:13]: Do you really open a server here?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+FROM golang:1.18.1
+WORKDIR /src
+
+COPY go.mod ./
+
+RUN go mod download
+
+COPY src/*.go ./
+
+RUN go build -o /docker-gs-ping
```
Comment (jeremyjams) [inline Dockerfile:11]:
```suggestion
RUN go build -o /nft-price-checker
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,44 @@
-# floor-price-checker
-Floor price checker
+# Golang OpenSea's NFT Collections floor price checker
++total portfolio value estimation
+
+### Build
+First, let's build the docker image
+```
+docker build . --tag floor-price-checker
```
Comment (jeremyjams) [inline README.md:7]: I would rename all `floor-price-checker` references to something more connected to your use-case, something like `nft-price-checker` or any suggestion (to find: meaningful and short) (we can rename the git repo to reflect that too)

File: README.md [hunk 5]
```diff
@@ -1,2 +1,44 @@
-# floor-price-checker
-Floor price checker
+# Golang OpenSea's NFT Collections floor price checker
```
Comment (jeremyjams) [inline README.md:1]: Good :)

File: src/app.go [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+	res, _ := http.DefaultClient.Do(req)
+	defer res.Body.Close()
+	body, _ := ioutil.ReadAll(res.Body)
+
+	json.Unmarshal(body, &col1)
+
+	sum += col1.Stats.Floor_price
+
+	if col1.Stats.Floor_price == 0 {
+		return collection_name + " est impossible a trouver sur Opensea"
```
Comment (jeremyjams) [inline src/app.go:59]: English is best :)

File: src/app.go [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+	var entries []string
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		entries = append(entries, scanner.Text())
+	}
+	return entries, scanner.Err()
+}
+
+// Asking for Opensea's colections Floor Prices
+func floorPrice(collection_name string) string {
```
Comment (jeremyjams) [inline src/app.go:44]: I would split `getFloorPrice` price fetching from the addition to the portfolio. What do you think?

File: Dockerfile (hunk 1 above)
Comment (mathisfchx) [reply Dockerfile:2]: Yes i'm gonna go with alpine version, it doesn't seem to make a change on the execution

File: Dockerfile (hunk 2 above)
Comment (mathisfchx) [reply Dockerfile:13]: oops bad understanding on go build options

File: Dockerfile (hunk 3 above)
Comment (mathisfchx) [reply Dockerfile:11]: yes definitly

File: README.md (hunk 4 above)
Comment (mathisfchx) [reply README.md:7]: Yes ```nft-price-checker``` seems fine

File: README.md (hunk 5 above)
Comment (mathisfchx) [reply README.md:1]: :)

File: src/app.go (hunk 6 above)
Comment (mathisfchx) [reply src/app.go:59]: Yes definitly

File: src/app.go (hunk 7 above)
Comment (mathisfchx) [reply src/app.go:44]: Yes it will be more understandable ! I'm looking at it right now

File: src/app.go [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+	if readErr != nil {
+		log.Fatal(readErr)
+	}
+
+	// Append some results in /iexec_out/
+	fr, err := os.OpenFile(iexec_out+"/result.txt", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, col := range entries {
```
Comment (jeremyjams) [inline src/app.go:93]: I think you can extract L93-L106 in a dedicated method in order to keep a lighter main

File: README.md (hunk 4 above)
Comment (jeremyjams) [inline README.md:7]:
```suggestion
docker build . --tag nft-price-checker
```

File: README.md [hunk 9]
```diff
@@ -1,2 +1,44 @@
...
+```
+rm -rf /tmp/iexec_out && \
+docker run \
+    -v /tmp/iexec_in:/iexec_in \
+    -v /tmp/iexec_out:/iexec_out \
+    -e IEXEC_IN=/iexec_in \
+    -e IEXEC_OUT=/iexec_out \
+    -e IEXEC_INPUT_FILE_NAME_1=input \
+    -e IEXEC_INPUT_FILES_NUMBER=1 \
+    floor-price-checker
```
Comment (jeremyjams) [inline README.md:35]:
```suggestion
    nft-price-checker
```

File: src/app.go [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+)
+
+var sum float64
```
Comment (jeremyjams) [inline src/app.go:13]:
```suggestion
```

File: src/app.go [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+
+	// Append some results in /iexec_out/
+	fr, err := os.OpenFile(iexec_out+"/result.txt", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	appendResult(fr, entries)
+
+	if _, err := fr.Write([]byte("------------- \n The estimate total value of your portfolio is : " + fmt.Sprintf("%f", sum) + " eth\n Or " + fmt.Sprintf("%f", sum*ethPrice()) + " Usd")); err != nil {
```
Comment (jeremyjams) [inline src/app.go:112]: You can move L112-L114 to `writeFloorPricesAndTotalValueToResult(..)` to keep a clean main :)

File: src/app.go [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+		log.Fatal(readErr)
+	}
+
+	// Append some results in /iexec_out/
+	fr, err := os.OpenFile(iexec_out+"/result.txt", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	appendResult(fr, entries)
```
Comment (jeremyjams) [inline src/app.go:110]:
```suggestion
	writeFloorPricesAndTotalValueToResult(fr, entries)
```

File: src/app.go (hunk 10 above)
Comment (mathisfchx) [reply src/app.go:13]: I do need the var sum to be global if i want to increment it outside of the main() : I do so in writeFloorPricesAndTotalValueToResult(). I can tweak it to work without a global variable by adding a return to this func, wich way do you this it the cleanest ?

Comment (jeremyjams) [reply src/app.go:13]: Does `sum` need to go outside `writeFloorPricesAndTotalValueToResult(..)`?

File: src/nft-price-checker.go [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+	for _, col := range entries {
+		fp := floorPrice(col)
+		sum += fp
+		if fp == 0 {
+			result += ("  x " + col + " cannot be found on Opensea" + "\n")
+		} else {
+			result += ("--> " + col + " Floor price = " + fmt.Sprintf("%f", fp) + " eth" + "\n")
+		}
+	}
+	result += ("------------- \n The estimate total value of your portfolio is : " + fmt.Sprintf("%f", sum) + " eth\n Or " + fmt.Sprintf("%f", sum*ethPrice()) + " Usd")
```
Comment (jeremyjams) [inline src/nft-price-checker.go:85]:
```
If sum > 0
result += "bla"
else
result += "blu"
```

**Review by james-toussaint — APPROVED** [review]
Congrat :clap:

**Pull request merged**